.github/workflows/ci_tests.yml

```diff
@@ -12,63 +12,90 @@ on:
-          - { id: ubuntu-clang-static-cxx20, platform: ubuntu, cc: clang, cpp: clang++, cmake_args: "-DCMAKE_CXX_STANDARD=20 -DCMAKE_CXX_STANDARD_REQUIRED=on"}
-          - { id: ubuntu-clang-static-cxx23, platform: ubuntu, cc: clang, cpp: clang++, cmake_args: "-DCMAKE_CXX_STANDARD=23 -DCMAKE_CXX_STANDARD_REQUIRED=on"}
-          - { id: ubuntu-clang-static-cxx26, platform: ubuntu, cc: clang, cpp: clang++, cmake_args: "-DCMAKE_CXX_STANDARD=26 -DCMAKE_CXX_STANDARD_REQUIRED=on"}
+        platform: [ubuntu-latest]
+        compiler:
+          - cpp: g++
+            c: gcc
+          - cpp: clang++
+            c: clang
+        cmake_args:
+          - description: "Static default"
+            args: ""
+          - description: "Dynamic default"
+            args: "-DBUILD_SHARED_LIBS=on"
+          - description: "static C++20"
+            args: "-DCMAKE_CXX_STANDARD=20 -DCMAKE_CXX_STANDARD_REQUIRED=on"
+          - description: "static C++23"
+            args: "-DCMAKE_CXX_STANDARD=23 -DCMAKE_CXX_STANDARD_REQUIRED=on"
+          - description: "static C++26"
+            args: "-DCMAKE_CXX_STANDARD=26 -DCMAKE_CXX_STANDARD_REQUIRED=on"
+        include:
+          - platform: ubuntu-latest
+            compiler:
+              cpp: g++
+              c: gcc
+            cmake_args:
+              description: "Werror"
+              cmake_args: "-DCMAKE_CXX_FLAGS='-Werror=all -Werror=extra'"
+          - platform: ubuntu-latest
+            compiler:
+              cpp: g++
+              c: gcc
+            cmake_args:
+              description: "A-San"
+              cmake_args: "-DCMAKE_CXX_FLAGS=-fsanitize=address -fsanitize=undefined"
+          - platform: ubuntu-latest
+            compiler:
+              cpp: g++
+              c: gcc
+            cmake_args:
+              description: "T-San"
+              cmake_args: "-DCMAKE_CXX_FLAGS=-fsanitize=thread"
 
+    name: "Bulid & Test: ${{ matrix.compiler.c }} ${{ matrix.cmake_args.description }}"
+    runs-on: ${{ matrix.platform }}
     steps:
       - uses: actions/checkout@v4
-      # GitHub runners have updated the Ubuntu Linux Kernel to use strong ASLR,
-      # but LLVM is not configured for this change, and thus the address
-      # sanitizer breaks.
-      #
-      # The next image is supposed to fix this, so if the Ubuntu image has been
-      # updated, this work around is no longer required.
-      - name: get runner image version
-        id: runner-image-version
+      - name: Install Ninja
+        uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: "~3.25.0"
+          ninjaVersion: "^1.11.1"
+      - name: Print installed softwares
         run: |
-          echo "image-version=$(echo $ImageVersion)" >> "$GITHUB_OUTPUT"
-        working-directory: .
-      - name: modify number of bits to use for ASLR entropy
-        if: ${{ steps.runner-image-version.outputs.ImageVersion }} == '20240310.1.0'
```

Comment:
we no longer have to care about this. Current image version has well past this post.

(Also this condition was erroneous).